### ISSUE TITLE
Title splits in ZDB

### ISSUE THREAD

**Issue by MiriKon** [hbz/laser-ygor#128]
Can be seen as a continuation of this https://github.com/openlibraryenvironment/gokb/issues/50#event-2324343465 closed GOKb ticket. As we have seen in several journals from the Beck Hochschulmodul KBART file, some of these titles will have to be split by the ZDB staff because in some instances there is only one ZDB record for a journal as well as its Beiheft while there should actually be two separate records. How can instances like this be highlighted as a "warning" in YGOR so that it will be possible to spot problematic cases such as this right away after the KBART has been processed via YGOR?

**Comment by darupp** [comment]
@MiriKon Can you give us a concrete example with ZDB Links and a file? So we can analyze how YGOR processes these requests at the moment and how we can improve it.

**Comment by MiriKon** [comment]
@darupp Maybe the examples here https://github.com/openlibraryenvironment/gokb/issues/50#event-2324343465 will suffice?

**Comment by philboeselager** [comment]
@MiriKon This is still an issue, right? In order to find a solution, we have to answer the question: which details tell you that there we need two ZDB records? The linked file in #50 (see https://github.com/openlibraryenvironment/gokb/files/3152744/EZB-HBZBE-01476_hbzKon_beck_2019_05_07.txt) shows: there are two lines, containing the same identifiers. This may include both, `print_identifier` and `zdb_id` (if existent). One of the two Kbart lines has a word like "Beiheft" or "Beilage" in the `publication_title`.

Are there some more such details? I've seen that the `title_url` may contain something like "...-Beil", but this is something very specific to the provider (Beck) I would guess.

Hence, the solution would be: If

1. there are two lines containing the same identifiers
2. and one of the two lines contains a word like "Beiheft" or "Beilage"

... then ... do what? Define the Ygor record as erroneous ("red" flag)? ( @darupp )

@MiriKon Do you know more librarian terms like "Beiheft" (also in other languages)?

**Comment by MiriKon** [comment]
@philboeselager  Yes - this is still an issue. According to the DNB guidelines https://www.dnb.de/DE/Professionell/Services/ISSN/issn_node.html;jsessionid=483FCB0B7AB2B163B3CF722C1F7E4E26.internet281#doc224512bodyText5 , supplements (Beihefte) can/should have individual ISSNs but as they need to apply for a new ISSN for the supplement, we have no way of making sure that all the supplements have their own ISSN. Therefore, we would propose to mark those titles in YGOR that share an identifier (ISSN oder ZDB) but only as a "warning". The titles should still be processed via YGOR and appear in the GOKb - where the titles are merged into one record if I'm not mistaken ( @darupp )? As a consequence, YGOR wouldn't have to filter for terms like "supplement" / "Beiheft" etc. and only for titles which share an identifier.

**Comment by philboeselager** [comment]
@darupp : MiriKon suggests to mark shared-ISSN records (e. g. Beihefte) as "warning", which is we would put them into the yellow container. That yellow container is labelled "Gültige Kbart Einträge ohne ZDB-Anreicherung" by now.

- Would you agree to put them into the yellow container?
- How would we label it then?

**Comment by philboeselager** [comment]
By 5a0242c84c6e388ca407e74904a3293fa4194220, all Kbart entries, including Beihefte, Beilagen etc., are processed and can be sent to the GOKb - no matter, if they share any identifiers or not. (Some do, some don't.) This is intended behaviour. In fact, there are some records processed as "red", but this is a result of the history event date format ("YYYY-YYYY").

The remaining question is if we should shared-ID Beihefte in the yellow container.

**Comment by darupp** [comment]
Unfortunately, they belong to the _red_ container, because the GOKb API will not identify them as seperate identities. @MiriKon  is correct: If all identifiers are the same (in this case: ISSN), the GOKb will match those titles, so that all shared ISSNs will be a single title in the GOKb. And we can't tweak the GOKb API, as matching titles via identifiers is a core concept of the GOKb.

How to solve this issue? In the best case, the provider will have to get seperate issns for supplements, as it is stated in the german issn rules @MiriKon quoted as well as in the international issn rules:

> When a continuing resource: (a) has regular supplements; or (b) has another continuing resource inserted into it; or (c) has sub-series; or (d) when  its  titles  consist  of  a  common  title  and  several  dependent  title  designations  and/or dependent titles;     then each of these continuing resources has its own ISSN and key title. 

When providers won't do this, we need at least the internal provider ID that is really unique. 

So in short

  * If there is no unique ID, the record should come into the red container and not send to the GOKb
  * If there is a unique provider ID in "title_id", the record shoul come into the yellow section and send to the GOKb. @hornmo We have to look into the GOKb API if it supports a disambiguation by provider ids
  * the yellow container should be labeled "Titles lacking imortant data" or so

**Comment by philboeselager** [comment]
@darupp I understand: if there is a unique ZDB id, other IDs may be non-explicit.

What about these cases:

- Multimedia und Recht (ZDB id `2081778-2`, no ISSN) vs.
- Multimedia und Recht Beilage (ZDB id `2081778-2` & ISSN `1434-596X`)

--> yellow container? both records?

- Foo and Bar (ISSN `1234-5678` and no ZDB id) vs.
- Foo and Bar Beilage (ISSN `1234-5678` and ZDB id 1234567-8)

--> yellow container? both records?

**Comment by darupp** [comment]
*Case 1:* _Multimedia und Recht_  --> Both records in the yellow container
*Case 2:* _Foo and Bar_ --> Both records in the yellow container

**Comment by darupp** [comment]
@MiriKon For clarification how to work in this cases I quote myself, because maybe it is lost in too much text: 

When providers will not [provide distinct issns for Supplements], we need at least a correct zdb_id. So the work is for the editor to do an zdb entry. Is there is no zdb id, the title can be imported with no ID, but further work on this title is not possible any more because he will never be matched again.

**Comment by philboeselager** [comment]
@darupp One question, hopefully the last one: one of the test packages (Beck) shows many printIdentifiers, but almost no onlineIdentifiers. So, a "distinct ISSN" regards to both, onlineIdentifier and printIdentifier, right?

**Comment by darupp** [comment]
As far as I know, distinct pISSN would be fine, at least as long as there are no other double identifiers. @hornmo Will titles be merged if there are distinct pISSNs, but the same eISSNs? Maybe you could document the matching procedure including the cascading per different identifiers/names somewhere in the wiki, so that coding against the API will be easier? https://github.com/openlibraryenvironment/gokb/wiki/Integration-APIs%3A-Telling-GOKb-about-new-or-corresponding-resources-and-local-identifiers

**Comment by philboeselager** [comment]
@hornmo @darupp I've created a [pull request](https://github.com/hbz/laser-ygor/pull/194) assuming that we can use both, `onlineIdentifier` and `printIdentifier`, as ISSN distinction fields. If anything changes about that, please mention me in this issue and re-open, if it has been closed in the meantime.
